### PR TITLE
Update Simple Icons To Latest Versions Using Actions

### LINKED ISSUE #4: Update Simple Icons To Latest Versions Using Actions
From the next major release of `simple-icons` (v11, releasing on May 26, 2024), We will begin removing third-party extensions from our README list that are not up to date with at least the previous major release.
For example, when v11 is released, we will remove any extensions that don't support v10.0.0 or higher.

## How do I keep my package up to date?
There are two methods that prove most popular among other maintainers. It is up to you which you implement.

### Using `@latest` from a CDN
Projects that use the CDN version of the project (jsDelivr/unpkg) can keep up to date simply by changing the version number in their code to @latest. That way - you're always using the most up to date version of the icon package.

### Running a weekly CRON
Many of our third party contributors make use of GitHub actions, and a weekly CRON job to query that the version number of the main package has changed, and then update and build their own package. A great example of this is [the DrawIO package by mondeja](https://github.com/mondeja/simple-icons-drawio/blob/b7110096c700106004ac1950f6c5c9f14ab3f218/.github/workflows/publish.yml). The main package is released consistently on a Sunday - so we recommend running your CRON job on a Monday or Tuesday.

---

When you've got your extension running one of the above methods, drop us a [Pull Request](https://github.com/simple-icons/simple-icons/compare) to update your version number on the README.

If you're in need of any support in implementing one of the above - please feel free to [start a discussion](https://github.com/simple-icons/simple-icons/discussions) or [ask us on Discord](https://discord.gg/vUXFa7t5xJ).

### PR DESCRIPTION
**Pull request by mackenly**
Closes #4 

This PR adds a GitHub Action to create the latest release every week from the main simple icons repo.

### CODE REVIEW

**Review by mondeja — COMMENTED** [review]
Hi, maintainer of simple-icons here :wave: I'll provide some help if you don't mind to create the release workflow.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+              uses: RadovanPelka/github-action-json@main
+              with:
+                path: 'template/manifest.json'
+
+            - name: Increment version number
+              run: |
+                VERSION=$(echo "${{ steps.packageJson.outputs.version }}")
+                IFS='.' read -ra VER <<< "$VERSION"
+                ((VER[2]++))
+                echo "::set-output name=NEW_VERSION::${VER[0]}.${VER[1]}.${VER[2]}"
```
[the remark this answers is not in the document]
Comment (mackenly) [reply .github/workflows/release.yml:28]: Hi, thank you for the feedback. This is very much a draft, but thank you for the feedback, especially on the depreciated command.

**Comment by mackenly** [comment]
The API endpoint for simple icons releases doesn't seem to include assets. https://api.github.com/repos/simple-icons/simple-icons/releases/latest Going to just download a zip of the master branch instead.

**Comment by mondeja** [comment]
> The API endpoint for simple icons releases doesn't seem to include assets. https://api.github.com/repos/simple-icons/simple-icons/releases/latest Going to just download a zip of the master branch instead.

You can download by version from tags using the url `https://github.com/simple-icons/simple-icons/archive/refs/tags/{version}.zip` or you can get the tarball and zipball urls from the endpoint that you've provided querying `tarball_url` and `zipball_url` fields.

**Comment by mackenly** [comment]
@mondeja  That's a much better approach, thanks! 🎉
Do you have any experience with permission errors when running Python scripts in actions? I've been unable to figure out how to give it permission to add, edit, and delete files. I've [opened a discussion here](https://github.com/orgs/community/discussions/77227) with more details and logs.

**Comment by mondeja** [comment]
> @mondeja That's a much better approach, thanks! 🎉 Do you have any experience with permission errors when running Python scripts in actions? I've been unable to figure out how to give it permission to add, edit, and delete files. I've [opened a discussion here](https://github.com/orgs/community/discussions/77227) with more details and logs.

Try `os.chmod(file, stat.S_IWRITE)`

**Comment by mackenly** [comment]
@mondeja The permissions error ended up being because I was running the script from the wrong directory using `python script/main.py`. Since the script navigates between the folders using `os.getcwd()` it wasn't able to replace the `script` from the path and was trying to delete/edit/create files that didn't exist. 

This should be ready to go. I've removed the on push trigger leaving just the cron and dispatch so there aren't any releases currently published. If there's nothing special that needs to be done as far as simple-icons, I'll merge and run it manually for this release just to make sure it works right on `main`. Thank you for your help!

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            - name: Install python packages
+              run: |
+                python -m pip install --upgrade pip
+                pip install -r requirements.txt
+
+            - name: Run script/main.py with the latest Simple Icons release
+              run: |
+                cd script
+                python main.py ./../simple-icons-${{ needs.version.outputs.LATEST_TAG_NAME }}/simple-icons-${{ needs.version.outputs.LATEST_TAG_NAME }} ${{ needs.version.outputs.LATEST_TAG_NAME }}
+                cd ..
```
Comment (mondeja) [inline .github/workflows/release.yml:56]: The working directory is not preserved between steps. Also, you can use [`working-directory`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsworking-directory):

```suggestion
              working-directory: scripts
              run: python main.py ./../simple-icons-${{ needs.version.outputs.LATEST_TAG_NAME }}/simple-icons-${{ needs.version.outputs.LATEST_TAG_NAME }} ${{ needs.version.outputs.LATEST_TAG_NAME }}
```

**Review by mondeja — CHANGES_REQUESTED** [review]
Thanks for your work. Just a minor change...

**Review by mondeja — APPROVED** [review]
Great! 🥇

I'll update the simple-icons version using this package on the dynamic badge of the [simple-icons README](https://github.com/simple-icons/simple-icons#third-party-extensions).

**Comment by mondeja** [comment]
Ups, seems that the directory is `script/`, not `scripts/` 😅

**Comment by mackenly** [comment]
All good. Btw the PR you linked is on Blazor, but this one is for Stream Deck.

Release successfully created: [https://github.com/mackenly/simple-icons-stream-deck/releases](https://github.com/mackenly/simple-icons-stream-deck/releases)

**Pull request merged**